### PR TITLE
Platform Admin Approves Pending Store

### PR DESCRIPTION
**Pull request by hmorri32**
#### What does  this PR do?
adds platform_admin boolean on user model. Namespace admin store routes, add routes depending on store status and delegate stores in store controller by using params. 
#### Where should the reviewer start?
routes -> admin::stores controller, 
#### How should this be manually tested?
RSPEC -> or log in as a platform admin and navigate around
#### What are the relevant story numbers?
#153568631
#### Questions:
  - Do Migrations Need to be ran?
YES
  - Do Environment Variables need to be set?
NO
  - Any other deploy steps?
NO
#### Pivotal hook 
- [(Finishes) #153568631]

### CODE REVIEW

File: app/views/users/stores/index.html.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
         </tr>
       </thead>
       <tbody>
-        <%= @stores.each do |store| %>
```
Comment (randy-darsh) [inline app/views/users/stores/index.html.erb:12]: nice catch

File: app/views/admin/dashboard/_nav.html.erb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <nav class="nav nav-tabs" id="myTab" role="tablist">
-  <%= link_to "View Orders", admin_dashboard_index_path, class: "nav-item nav-link active", id: "nav-home-tab" %>
-    <%= link_to "View Items", admin_items_path, class: "nav-item nav-link" %>
-    <%= link_to "Update Account", edit_user_path(current_user), class: "nav-item nav-link" %>
-    <%= link_to "View Analytics", admin_analytics_path, class: "nav-item nav-link" %>
+  <%= link_to "View Orders",    admin_dashboard_index_path,   class: "nav-item nav-link active", id: "nav-home-tab" %>
+  <%= link_to "View Items",     admin_items_path,             class: "nav-item nav-link" %>
+  <%= link_to "Update Account", edit_user_path(current_user), class: "nav-item nav-link" %>
+  <%= link_to "View Analytics", admin_analytics_path,         class: "nav-item nav-link" %>
+  <%= link_to "Stores",         admin_stores_path,            class: 'nav-item nav-link' %>
```
Comment (randy-darsh) [inline app/views/admin/dashboard/_nav.html.erb:6]: looks nice and clean

File: app/views/users/stores/index.html.erb (hunk 1 above)
Comment (hmorri32) [reply app/views/users/stores/index.html.erb:12]: thx

File: spec/factories/users.rb [hunk 3]
```diff
@@ -13,4 +13,13 @@
     role       "admin"
     email { Faker::Internet.email }
   end
+
+  factory :platform_admin, class: User do
+    first_name { Faker::Hipster.name }
+    last_name  { Faker::AquaTeenHungerForce.character }
+    password   { Faker::SiliconValley.company }
+    email      { Faker::SiliconValley.app }
+    role       "admin"
```
Comment (randy-darsh) [inline spec/factories/users.rb:22]: faker ftw

**Comment by randy-darsh** [comment]
looks good. merging

File: app/controllers/admin/stores_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class Admin::StoresController < ApplicationController
+  def index
+    @stores = delegate_stores
+  end
+
+  def show
+    @store = Store.find(params[:id])
+  end
+
+  def delegate_stores
```
Comment (liambarstad) [inline app/controllers/admin/stores_controller.rb:10]: can this be a private method? also you can do something like "return Store.where(params[:status] ? status: params[:status])" (Store.where is the same as Store.all). I personally am not the biggest fan of switching a function by the value, but thats just me

File: app/views/admin/dashboard/_nav.html.erb (hunk 2 above)
Comment (liambarstad) [reply app/views/admin/dashboard/_nav.html.erb:6]: its so squeaky

**Pull request merged**